engine/backends/fstar/fstar_backend.ml

```diff
@@ -971,10 +994,7 @@ struct
                             |> List.map ~f:to_pattern) ),
                    ty );
                ] )
-    | Type { name; generics; _ }
-      when Attrs.find_unique_attr e.attrs
-             ~f:([%eq: Types.ha_payload] OpaqueType >> Fn.flip Option.some_if ())
-           |> Option.is_some ->
+    | Type { name; generics; _ } when is_erased ->
         if not ctx.interface_mode then
           Error.raise
           @@ {
```

Comment:
We should not error out here now: we should allow type definition to be opaque, whether we are producing a fsti, a fst or both